frontend/src/metabase/public/containers/PublicOrEmbeddedDashboard/PublicOrEmbeddedDashboardView.stories.tsx

```diff
@@ -0,0 +1,227 @@
+// @ts-expect-error There is no type definition
+import createAsyncCallback from "@loki/create-async-callback";
+import type { ComponentStory, Story } from "@storybook/react";
+import { useEffect, type ComponentProps } from "react";
+import { Provider } from "react-redux";
+
+import { getStore } from "__support__/entities-store";
+import { getNextId } from "__support__/utils";
+import { NumberColumn, StringColumn } from "__support__/visualizations";
+import { explicitSizeRefreshModeContext } from "metabase/components/ExplicitSize/context";
+import { publicReducers } from "metabase/reducers-public";
+import TABLE_RAW_SERIES from "metabase/visualizations/components/TableSimple/stories-data/table-simple-orders-with-people.json";
+import {
+  createMockCard,
+  createMockColumn,
+  createMockDashboard,
+  createMockDashboardCard,
+  createMockDataset,
+  createMockDatasetData,
+  createMockParameter,
+} from "metabase-types/api/mocks";
+import {
+  createMockDashboardState,
+  createMockState,
+} from "metabase-types/store/mocks";
+
+import { PublicOrEmbeddedDashboardView } from "./PublicOrEmbeddedDashboardView";
+
+export default {
+  title: "embed/PublicOrEmbeddedDashboardView",
+  component: PublicOrEmbeddedDashboardView,
+  decorators: [
+    ReduxDecorator,
+    FasterExplicitSizeUpdateDecorator,
+    WaitForResizeToStopDecorator,
+  ],
+};
+
+function ReduxDecorator(Story: Story) {
+  return (
+    <Provider store={store}>
+      <Story />
+    </Provider>
+  );
+}
+
+function FasterExplicitSizeUpdateDecorator(Story: Story) {
+  return (
+    <explicitSizeRefreshModeContext.Provider value="none">
+      <Story />
+    </explicitSizeRefreshModeContext.Provider>
+  );
+}
+
+/**
+ * This is an arbitrary number, it should be big enough to pass CI tests.
+ * This value works together with FasterExplicitSizeUpdateDecorator which
+ * make sure we finish resizing any ExplicitSize components the fastest.
+ */
+const TIME_UNTIL_ALL_ELEMENTS_STOP_RESIZING = 1000;
+function WaitForResizeToStopDecorator(Story: Story) {
+  const asyncCallback = createAsyncCallback();
+  useEffect(() => {
+    setTimeout(asyncCallback, TIME_UNTIL_ALL_ELEMENTS_STOP_RESIZING);
+  }, [asyncCallback]);
+
+  return <Story />;
+}
+
+const DASHBOARD_ID = getNextId();
```

Comment:
You don't have to focus too much on the mocked data, I just try mocking some charts for the dashboard.